tests/boot/grub_test_snapshot.pm

```diff
@@ -24,6 +24,13 @@ sub run {
     my $self = shift;
 
     select_console 'root-console';
+    # We need to delete snappers that were created during the zypper actions before boot into snapshot.
+    my $ret     = script_run("snapper --help | grep disable-used-space");
+    my $disable = '';
+    $disable = '--disable-used-space' unless $ret;
+    my @snaps   = split(/\n/, script_output("snapper list $disable | grep -vEw 'important=yes|single'| grep -w root | awk -F '|' '{print \$1}'"));
+    my $id_list = join(' ', @snaps);
+    assert_script_run("snapper delete $id_list", 300);
```

Comment:
I think even it deleted failed, it won't impact our test flow. Maybe at here, we can use 'script_run'

Reply:
changed and also added the checking for  if  @snaps is null